nilearn/_utils/tests/test_numpy_conversions.py

```diff
@@ -126,25 +125,30 @@ def test_as_ndarray():
         (bool, "F", True, np.int8, None, True),
         (bool, "C", True, np.int8, "C", True),
         (bool, "F", True, np.int8, "F", True),
-    ]
-
+    ],
+)
+def test_as_ndarray(in_dtype, in_order, copy, out_dtype, out_order, copied):
+    # All test cases
+    # input dtype, input order, should copy, output dtype, output order, copied
     shape = (10, 11)
-    for case in test_cases:
-        in_dtype, in_order, copy, out_dtype, out_order, copied = case
-        arr1 = np.ones(shape, dtype=in_dtype, order=in_order)
-        arr2 = as_ndarray(arr1, copy=copy, dtype=out_dtype, order=out_order)
-        assert not are_arrays_identical(arr1[0], arr2[0]) == copied, str(case)
-        if out_dtype is None:
-            assert arr2.dtype == in_dtype, str(case)
-        else:
-            assert arr2.dtype == out_dtype, str(case)
-
-        result_order = out_order if out_order is not None else in_order
-        if result_order == "F":
-            assert arr2.flags["F_CONTIGUOUS"], str(case)
-        else:
-            assert arr2.flags["C_CONTIGUOUS"], str(case)
-
+    # for case in test_cases:
```

Comment:
to be cleaned

Reply:
tried to use more transparent variable names and add comment to explain what is happening